dfdaemon/config/config.go

```diff
@@ -127,30 +134,47 @@ func (p *Properties) Validate() error {
 		)
 	}
 
-	if ok, _ := regexp.MatchString("^[[:digit:]]+[MK]$", p.RateLimit); !ok {
-		return dferr.Newf(
-			constant.CodeExitRateLimitInvalid,
-			"invalid rate limit %s", p.RateLimit,
-		)
-	}
-
 	return nil
 }
 
 // DFGetConfig returns config for dfget downloader
-func (p *Properties) DFGetConfig() DFGetConfig {
+func (p *Properties) DFGetConfig(conf *config.Properties) DFGetConfig {
 	// init DfgetFlags
-	var dfgetFlags []string
-	dfgetFlags = append(dfgetFlags, p.DfgetFlags...)
-	dfgetFlags = append(dfgetFlags, "--dfdaemon")
+	dfgetFlags := convertDFGetConfigToArgs(conf)
 	if p.Verbose {
 		dfgetFlags = append(dfgetFlags, "--verbose")
 	}
 
+	// if set supernodes in dfdaemon, dfget will use them,
+	// otherwise dfget will use the supernodes in config.
+	if len(p.SuperNodes) > 0 {
+		dfgetFlags = append(dfgetFlags, "--node", strings.Join(p.SuperNodes, ","))
+	} else if conf != nil && len(conf.Nodes) > 0 {
+		dfgetFlags = append(dfgetFlags, "--node", strings.Join(conf.Nodes, ","))
+	}
+
+	// If don't set limit in dfget config, use dfdaemon rate limit.
+	if p.RateLimit != 0 {
```

Comment:
Consider encapsulating  it as a function just like this

```
func (p *Properties) parseLimit(conf *config.Properties) []string {
	dfgetFlags := make([]string, 0)
	if conf == nil {
		if p.RateLimit != 0 {
			dfgetFlags = append(dfgetFlags, "--locallimit", p.RateLimit.String())
			dfgetFlags = append(dfgetFlags, "--totallimit", p.RateLimit.String())
			return dfgetFlags
		}
		rate := netutils.NetLimit()
		dfgetFlags = append(dfgetFlags, "--locallimit", rate.String())
		dfgetFlags = append(dfgetFlags, "--totallimit", rate.String())
		return dfgetFlags
	}

	if p.RateLimit == 0 {
		return nil
	}

	if conf.LocalLimit == 0 {
		dfgetFlags = append(dfgetFlags, "--locallimit", p.RateLimit.String())
	}
	if conf.TotalLimit == 0 {
		dfgetFlags = append(dfgetFlags, "--totallimit", p.RateLimit.String())
	}
	return dfgetFlags
}
```

Reply:
Done